File: app/src/main/java/com/merino/ddfilms/model/MovieDetails.java
package com.merino.ddfilms.model;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

import java.util.List;

import lombok.Data;

@Data
public class MovieDetails implements Parcelable {
    private int id;
    private String title;
    private String overview;
    @SerializedName("release_date")
    private String releaseDate;
    private int runtime;
    private String tagline;
    private String status;
    @SerializedName("original_language")
    private String originalLanguage;
    @SerializedName("backdrop_path")
    private String backdropPath;
    @SerializedName("poster_path")
    private String posterPath;
    @SerializedName("vote_average")
    private double voteAverage;
    @SerializedName("vote_count")
    private int voteCount;
    private List<Genre> genres;

    protected MovieDetails(Parcel in) {
        id = in.readInt();
        title = in.readString();
        overview = in.readString();
        releaseDate = in.readString();
        runtime = in.readInt();
        tagline = in.readString();
        status = in.readString();
        originalLanguage = in.readString();
        backdropPath = in.readString();
        posterPath = in.readString();
        voteAverage = in.readDouble();
        voteCount = in.readInt();
        genres = in.createTypedArrayList(Genre.CREATOR);
    }



    public static final Creator<MovieDetails> CREATOR = new Creator<MovieDetails>() {
        @Override
        public MovieDetails createFromParcel(Parcel in) {
            return new MovieDetails(in);
        }

        @Override
        public MovieDetails[] newArray(int size) {
            return new MovieDetails[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeInt(id);
        parcel.writeString(title);
        parcel.writeString(overview);
        parcel.writeString(releaseDate);
        parcel.writeInt(runtime);
        parcel.writeString(tagline);
        parcel.writeString(status);
        parcel.writeString(originalLanguage);
        parcel.writeString(backdropPath);
        parcel.writeString(posterPath);
        parcel.writeDouble(voteAverage);
        parcel.writeInt(voteCount);
        parcel.writeTypedList(genres);
    }

    public String getDuration() {
        int hours = runtime / 60;
        int minutes = runtime % 60;
        return hours > 0 ? hours + "h " + minutes + "m" : minutes + "m";
    }

    public static class Genre implements Parcelable {
        private int id;
        private String name;

        protected Genre(Parcel in) {
            id = in.readInt();
            name = in.readString();
        }

        public static final Creator<Genre> CREATOR = new Creator<Genre>() {
            @Override
            public Genre createFromParcel(Parcel in) {
                return new Genre(in);
            }

            @Override
            public Genre[] newArray(int size) {
                return new Genre[size];
            }
        };

        @Override
        public void writeToParcel(Parcel parcel, int i) {
            parcel.writeInt(id);
            parcel.writeString(name);
        }

        @Override
        public int describeContents() {
            return 0;
        }
    }
}

File: app/src/main/java/com/merino/ddfilms/model/MovieDetails.kt
package com.merino.ddfilms.model

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize

@Parcelize
data class MovieDetails(
    var id: Int = 0,
    var title: String? = null,
    var overview: String? = null,
    @SerializedName("release_date") var releaseDate: String? = null,
    var runtime: Int = 0,
    var tagline: String? = null,
    var status: String? = null,
    @SerializedName("original_language") var originalLanguage: String? = null,
    @SerializedName("backdrop_path") var backdropPath: String? = null,
    @SerializedName("poster_path") var posterPath: String? = null,
    @SerializedName("vote_average") var voteAverage: Double = 0.0,
    @SerializedName("vote_count") var voteCount: Int = 0,
    var genres: List<Genre>? = null
) : Parcelable {

    fun getDuration(): String {
        val hours = runtime / 60
        val minutes = runtime % 60
        return if (hours > 0) "${hours}h ${minutes}m" else "${minutes}m"
    }

    @Parcelize
    data class Genre(
        var id: Int = 0,
        var name: String? = null
    ) : Parcelable
}
